### PR TITLE
[MB-8335] upload admended orders part2 (API changes)

### PR DESCRIPTION
**Pull request by jacquelineIO**
## Description

Making a lot of changes to https://github.com/transcom/mymove/pull/6870 and want to make sure I'm going in the right direction so I created a new branch and PR to review this work.

## Reviewer Notes

Is there anything you would like reviewers to give additional scrutiny?

## Setup

Add any steps or code to run in this section to help others prepare to run your code:

```sh
echo "Code goes here"
```

## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely) have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](tbd) for this change
* [this article](tbd) explains more about the approach used.

## Screenshots

If this PR makes visible UI changes, an image of the finished UI can help reviewers and casual
observers understand the context of the changes. A before image is optional and
can be included at the submitter's discretion.

Consider using an animated image to show an entire workflow instead of using multiple images. You may want to use GIPHY CAPTURE for this! 📸

_Please frame screenshots to show enough useful context but also highlight the affected regions._

### CODE REVIEW

File: pkg/uploader/uploader.go [hunk 1]
```diff
@@ -34,6 +54,12 @@ var ErrFileSizeLimitExceedsMax = errors.Errorf("FileSizeLimit exceeds max of %d
 // file of any size the file will be locked from downloading forever.
 const MaxFileSizeLimit = 250 * MB
 
+// MaxUserFileSizeLimit sets the maximum file size limit
+// Anti-Virus scanning won't be able to scan files larger than 250MB
+// Any unscanned files will not be available for download so while we can upload a larger
+// file of any size the file will be locked from downloading forever.
+const MaxUserFileSizeLimit = 25 * MB
```
Comment (mboren) [inline pkg/uploader/uploader.go:61]: nit: comment says 250MB, code says 25MB

Comment (jacquelineIO) [reply pkg/uploader/uploader.go:61]: @mboren yes, I need to fix this... not sure where I landed with this, but something in this area will need to be updated and made consistent. https://ustcdp3.slack.com/archives/CP6PTUPQF/p1625001310033000

**Comment by duncan-truss** [comment]
Can confirm I was able to upload amended orders and the TOO was alerted and asked to acknowledge them. 🎉 

@mboren Do you know the plan for the move status being changed from approved to approvals requested in the interim?  That's the one issue I had with editing the orders as the TOO is the status still being in Approved.  I could loosen the restriction if we can't fit that work in.  I know Hanna's implementation was around the customer submitting the move again not just saving the uploaded orders.  Happy to follow up with Pamplemoose if that is their concern now.

Screen Shot 2021-07-01 at 6 07 42 PM
Screen Shot 2021-07-01 at 6 08 13 PM

File: pkg/services/order/order_updater.go [hunk 2]
```diff
@@ -183,6 +192,68 @@ func orderFromTOOPayload(existingOrder models.Order, payload ghcmessages.UpdateO
...
 
+func (f *orderUpdater) amendedOrder(db *pop.Connection, logger services.Logger, userID uuid.UUID, order models.Order, file io.ReadCloser, filename string, storer storage.FileStorer) (models.UserUpload, string, error) {
+
+	// If Order does not have a Document for amended orders uploads, then create a new one
+	var err error
+	savedAmendedOrdersDoc := order.UploadedAmendedOrders
+	if order.UploadedAmendedOrders == nil {
+		amendedOrdersDoc := &models.Document{
+			ServiceMemberID: order.ServiceMemberID,
+			CreatedAt:       time.Now(),
```
Comment (duncan-truss) [inline pkg/services/order/order_updater.go:203]: I think pop will automatically set CreatedAt for us right?

**Comment by mboren** [comment]
@duncan-truss Moose is working on changing the move status over in this PR: https://github.com/transcom/mymove/pull/6897/

File: pkg/testdatagen/shared.go [hunk 3]
```diff
@@ -199,6 +203,38 @@ func Fixture(name string) afero.File {
...
 
+// FixtureRuntimeFile allows us to include a fixture like a PDF in the test
+func FixtureRuntimeFile(name string) *runtime.File {
+	fixtureDir := "testdatagen/testdata"
+	cwd, err := os.Getwd()
+	if err != nil {
+		log.Panic(err)
+	}
+
+	fixturePath := path.Join(cwd, "..", "..", fixtureDir, name)
```
Comment (duncan-truss) [inline pkg/testdatagen/shared.go:214]: Ha I had to do something similar in the function above, not sure if we have a preference for allowing it to be called from outside pkg/testdatagen.

https://github.com/transcom/mymove/pull/6908/files#diff-f17bd29de6bf79d93c075007ecb5acffb23cc20e7a9ebaf7ce4ea64f0cf2c762R195-R197

File: pkg/services/order/order_updater.go (hunk 2 above)
Comment (jacquelineIO) [reply pkg/services/order/order_updater.go:203]: @duncan-truss yes. I will update.

File: pkg/services/order/order_updater.go [hunk 4]
```diff
@@ -95,29 +97,36 @@ func (f *orderUpdater) UpdateAllowanceAsCounselor(orderID uuid.UUID, payload ghc
...
+			return nil, services.NewNotFoundError(orderID, "while looking for order")
+		}
+	}
 
+	return &order, nil
+}
+
+func (f *orderUpdater) findOrderWithAmendedOrders(orderID uuid.UUID) (*models.Order, error) {
+	var order models.Order
+	err := f.db.Q().EagerPreload("Moves", "ServiceMember", "Entitlement", "UploadedAmendedOrders").Find(&order, orderID)
```
Comment (duncan-truss) [inline pkg/services/order/order_updater.go:129]: I'm wondering what is really required here to preload besides UploadedAmendedOrders. I don't think we're using any of this data but I know we also want to preserve what associations values already exist in the database to not remove them when the order model gets updated.

Comment (jacquelineIO) [reply pkg/services/order/order_updater.go:129]: I don't even this any of this is passed up from this specific all. I'm just looking for `UploadedAmendedOrders`. The other `findOrders` function is used for other stuff you mentioned, so I should just test it with none of the other stuff.

**Review by duncan-truss — APPROVED** [review]

**Pull request merged**